### PR TITLE
Add Spec Coverage for ExportCodeInput screen.

### PR DESCRIPTION
**Pull request by mattThousand**
### Why:
Currently the feature of enabling/disabling the `Next` button on the
`ExportCodeInput` screen depending on the length of user input does not have test coverage. As this is an important part of the user flow, we would like to add some basic test coverage.

### This Commit
This commit adds tests for the above, and also converts related files to TypeScript.

85642488-a7b49700-b65f-11ea-92f2-39bc50c4bf1b

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1274,7 +1274,6 @@
 "@testing-library/jest-native@^3.1.0":
   version "3.1.0"
   resolved "https://registry.yarnpkg.com/@testing-library/jest-native/-/jest-native-3.1.0.tgz#414c834dfb524ca82b86509d5479e365d4d5bbec"
-  integrity sha512-MZdsTsD6xrP5V0FTz2gVPG00ZMoKn4WniRitUH40zZTQ0DfMa5CVDpkAd5ed6qX1coDwS8vSCZS/JGMIDYjRQQ==
```
Comment (octohub) [inline yarn.lock]: Should these integrity values be removed?

**Review by octohub — APPROVED** [review]
Looks solid, I would like @johnschoeman to take a look as well given his domain expertise.

File: app/views/__tests__/ExportCodeInput.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import {
+  NavigationScreenProp,
+  NavigationRoute,
+  NavigationState,
+  NavigationParams,
+} from 'react-navigation';
+import { render, cleanup } from '@testing-library/react-native';
+import '@testing-library/jest-native/extend-expect';
+import { fireEvent } from '@testing-library/react-native';
+import { TracingStrategy } from '../../../app/tracingStrategy';
```
Comment (johnschoeman) [inline app/views/__tests__/ExportCodeInput.spec.tsx]: I might group these imports such that third party code is separated from our code by a line break

File: app/views/__tests__/ExportCodeInput.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+interface NavProps {
+  navigation: NavigationScreenProp<NavigationState, NavigationParams>;
+  route: NavigationRoute;
+}
+
+afterEach(cleanup);
+
+describe('<ExportCodeInputScreen />', () => {
+  beforeEach(() => {});
```
Comment (johnschoeman) [inline app/views/__tests__/ExportCodeInput.spec.tsx]: is this beforeEach necessary?

File: app/views/__tests__/ExportCodeInput.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  it('when there is no text input', () => {
```
Comment (johnschoeman) [inline app/views/__tests__/ExportCodeInput.spec.tsx:34]:
```suggestion
  describe('when there is no text input', () => {
     it('renders with the next button disabled', () => {
     
```

File: app/views/__tests__/ExportCodeInput.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    for (let i = 0; i <= 5; i++) {
+      const input = getByTestId(`input${i}`);
+      fireEvent.changeText(input, '1');
+    }
+    const button = getByTestId('next-button');
+    expect(button).not.toBeDisabled();
+  });
+});
+
+const renderTracingStrategyProvider = (strategy: TracingStrategy) => {
```
Comment (johnschoeman) [inline app/views/__tests__/ExportCodeInput.spec.tsx:70]: This totally works, but is a bit of a smell to me. I feel like the tests would be better Unit tests if the CodeInput was less couple to the data model and we could pass in all of the data required as a prop.  so basically decouple the notion of a `Screen` which know how to handle navigation and application IO stuff and a `CodeInput` which know how to validate code inputs from a user.

**Review by johnschoeman — APPROVED** [review]
LGTM, a few comments

File: app/views/__tests__/ExportCodeInput.spec.tsx (hunk 5 above)
Comment (mattThousand) [reply app/views/__tests__/ExportCodeInput.spec.tsx:70]: I think I follow but just want to make sure; are you envisioning 2 separate components, an `ExportScreen` and a `CodeInput`?

Comment (johnschoeman) [reply app/views/__tests__/ExportCodeInput.spec.tsx:70]: yup something like that! I wouldnt say that is necessary for this commit, but I think that would a way to make things a bit more modular.

**Pull request closed without merging**